smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -39,6 +41,7 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
             // only escape words when the symbol has a definition file to prevent escaping intentional references to built-in shapes
             .escapePredicate { _, symbol -> symbol.definitionFile.isNotEmpty() }
             .buildEscaper()
+        nullableIndex = NullableIndex(model)
```

Comment:
**Nit**: There's no need to set this in the `init` block. It can be set directly after the field definition on L26:

```kotlin
private val nullableIndex = NullableIndex(model)
```

Reply:
Make sense, I will change it accordingly.